graylog2-web-interface/src/components/users/UserCreate/UserCreate.jsx

```diff
@@ -88,10 +106,47 @@ const UserCreate = () => {
   const _handleCancel = () => history.push(Routes.SYSTEM.USERS.OVERVIEW);
   const hasValidRole = selectedRoles.size > 0 && selectedRoles.filter((role) => role.name === 'Reader' || role.name === 'Admin');
 
+  const getUserNameGroup = () => {
+    if (isCloud) {
+      const cloudPlugin = PluginStore.exports('cloud');
+      const CloudUserNameFormGroup = cloudPlugin?.[0]?.CloudUserNameFormGroup;
+
+      return <CloudUserNameFormGroup />;
+    }
+
+    return (
+      <>
+        <UsernameFormGroup users={users} />
+        <EmailFormGroup />
+      </>
+    );
+  };
+
+  const getPasswordGroup = () => {
+    if (isCloud) {
+      const cloudPlugin = PluginStore.exports('cloud');
+      const CloudPasswordFormGroup = cloudPlugin?.[0]?.CloudPasswordFormGroup;
+
+      return <CloudPasswordFormGroup />;
+    }
+
+    return <PasswordFormGroup />;
+  };
+
   if (!users) {
     return <Spinner />;
   }
 
+  const showSubmitError = (errors) => {
+    if (isCloud) {
+      const oktaError = JSON.parse(errors?.additional?.res?.text);
+
+      return oktaError && oktaError?.message;
```

Comment:
I think this logic should live in the plugin instead of here.

Reply:
Looking at the error message, I also wonder if that error should really be exposed in here.


Toastr notification:
![error](https://user-images.githubusercontent.com/716185/103923338-45ac1280-5115-11eb-8446-cc6d8dba9c36.png)

Error in form:
![error2](https://user-images.githubusercontent.com/716185/103923344-46dd3f80-5115-11eb-9bf7-9047cda9abdc.png)

I think we should try to extract the Okta API message, since that is what I would expect to see. Looking at the response this may be possible in the server, but I think it's worth exploring it. Do you think that would be possible @danotorrey?

Reply:
I created https://github.com/Graylog2/graylog2-server/issues/9884 to document this issue, so we can move on with this.